### PR TITLE
Fix determine segment rearchive date

### PR DESCRIPTION
**Pull request by sgiehl**
### Review

* [ ] Functional review done
* [ ] Potential edge cases thought about (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] Usability review done (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] Security review done [see checklist](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] Code review done
* [ ] Tests were added if useful/possible
* [ ] Reviewed for breaking changes
* [ ] Developer changelog updated if needed
* [ ] Documentation added if needed
* [ ] Existing documentation updated if needed

### CODE REVIEW

File: core/CronArchive/SegmentArchiving.php [hunk 1]
```diff
@@ -111,16 +111,25 @@ public function getReArchiveSegmentStartDate($segmentInfo)
...
+                return null;
+            }
             $this->logger->debug("process_new_segments_from set to segment_last_edit_time, segment last edit time is {time}",
                 array('time' => $segmentLastEditedTime));
 
             return $segmentLastEditedTime;
-        } else if (preg_match("/^editLast([0-9]+)$/", $this->processNewSegmentsFrom, $matches) && !empty($segmentLastEditedTime)) {
+        } else if (preg_match("/^editLast([0-9]+)$/", $this->processNewSegmentsFrom, $matches)) {
+            if (empty($segmentLastEditedTime)) {
+                return null;
```
Comment (tsteur) [inline core/CronArchive/SegmentArchiving.php:131]: was going to say here we could maybe fall back to `Date::now()` but I suppose it's not really needed anyway as we should always have an edited time.

**Review by tsteur — COMMENTED** [review]
LGTM. Left one comment to look at optionally but not too important.

File: core/CronArchive/SegmentArchiving.php (hunk 1 above)
Comment (sgiehl) [reply core/CronArchive/SegmentArchiving.php:131]: An empty edit time shouldn't happen, indeed. Nevertheless assuming `Date::now()` when it wouldn't be set for any reason would only be correct if we assume this method is only called when the segment had been edited/created.

Comment (tsteur) [reply core/CronArchive/SegmentArchiving.php:131]: 👍  all good, it's not needed. LGTM

**Pull request merged**